client/resourcemanager_client.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pd
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/errors"
+	rmpb "github.com/pingcap/kvproto/pkg/resource_manager"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+type actionType int
+
+const (
+	add    actionType = 0
+	modify actionType = 1
+)
+
+// ResourceManagerClient manages resource group info and token request.
+type ResourceManagerClient interface {
+	ListResourceGroups(ctx context.Context) ([]*rmpb.ResourceGroup, error)
+	GetResourceGroup(ctx context.Context, resourceGroupName string) (*rmpb.ResourceGroup, error)
+	AddResourceGroup(ctx context.Context, metaGroup *rmpb.ResourceGroup) (string, error)
+	ModifyResourceGroup(ctx context.Context, metaGroup *rmpb.ResourceGroup) (string, error)
+	DeleteResourceGroup(ctx context.Context, resourceGroupName string) (string, error)
+	AcquireTokenBuckets(ctx context.Context, request *rmpb.TokenBucketsRequest) ([]*rmpb.TokenBucketResponse, error)
+}
+
+// resouceManagerClient gets the ResourceManager client of current PD leader.
+func (c *client) resouceManagerClient() rmpb.ResourceManagerClient {
+	if cc, ok := c.clientConns.Load(c.GetLeaderAddr()); ok {
+		return rmpb.NewResourceManagerClient(cc.(*grpc.ClientConn))
+	}
+	return nil
+}
+
+// ListResourceGroups loads and returns target keyspace's metadata.
```

Comment:
The comment is incorrect.

```suggestion
// ListResourceGroups loads and returns xxx.
```